tensorflow/core/kernels/argmax_op.cc

```diff
@@ -57,27 +57,29 @@ class ArgOp : public OpKernel {
     const int32 dim = internal::SubtleMustCopy(dimension.scalar<int32>()());
     const int input_dims = input.dims();
 
-    OP_REQUIRES(context, dim >= 0, errors::InvalidArgument("dim must be >= 0"));
-    OP_REQUIRES(context, dim < input_dims,
-                errors::InvalidArgument("Minimum tensor rank: ", dim + 1,
+    int axis = dim < 0 ? dim + input_dims : dim;
+
+    OP_REQUIRES(context, axis >= 0, errors::InvalidArgument("axis must be >= 0"));
```

Comment:
It would be worth adding more details to this error message, since `axis` is not a variable that the user has input. Surfacing both `dim` and `input_dim` in the message would be helpful for people to debug.